scripts/CodeCoverage.py

```diff
@@ -0,0 +1,189 @@
+    try:
+        ret_code = subprocess.check_call([GCOVR_CMD_STR], stderr=subprocess.STDOUT, shell=True)
+    except:
+        print "Gcovr return code: bad"
+
+def copy_coverage_files_into_cov_dir(object_dir, launch_dir):
+    CP_COV_FILES_STR = 'cp ' + launch_dir + '/build/CMakeFiles/' + object_dir + '/src/* ' + launch_dir +'/coverage'
+    try:
+        ret_code = subprocess.check_call([CP_COV_FILES_STR], stderr=subprocess.STDOUT, shell=True)
+    except:
+        print "Copy coverage files return code: bad"
+
+def format_user_list(user_list):
+    formatted_list = None
+    if user_list is not None:
+        for whitelist_item in user_list.split():
+            print "Split into : " + whitelist_item
+            if formatted_list is None:
+                formatted_list = '.*' + whitelist_item + '$'
+            else:
+                formatted_list = formatted_list + '|.*'+whitelist_item+'$'
+    else:
+        print "list is empty" #Write something better here
+    return formatted_list
+
+def generate_gcovr_filter(formatted_user_list, default_list):
+    if formatted_user_list is None:
+        return default_list
+    else:
+        return default_list + '|' + formatted_user_list
+
+def main(argv):
+    argParser = argparse.ArgumentParser(description="Generate code coverage information for a Network Monitor repository that uses CMake")
+    argParser.add_argument("-w", "--whitelist", dest="whitelist", metavar="HEADER_FILE", help="Whitelist a header or template file for code coverage")
+    argParser.add_argument("-b", "--blacklist", dest="blacklist", metavar="SOURCE_FILE", help="Blacklist a source file for code coverage")
+   
+    args = argParser.parse_args()
+
+    santize_input_args(arg_parser=argParser, args=args)
+  
+    found_project_name, project_name = get_project_name_from_CMakeLists_file(full_file_path=CMAKEFILEPATH)
+    if not found_project_name:
+        print "ERROR: No project name found in CMakeLists.txt"
+        sys.exit(2)   
+
+    LAUNCH_DIR = os.getcwd()
+    PROJECT = project_name
+    OBJ_DIR = PROJECT + '.dir'
+    USER_WHITELIST = None
+    USER_BLACKLIST = None
+    GTEST_ZIP_PATH = LAUNCH_DIR + '/3rdparty/gtest-1.7.0.zip'
+
+    if args.whitelist:
+        USER_WHITELIST = args.whitelist
+        print "USER_WHITELIST = " + USER_WHITELIST
+    if args.blacklist:
+        USER_BLACKLIST = args.blacklist
+        print "USER_BLACKLIST = " + USER_BLACKLIST
+
+    unzip_file(full_file_path=GTEST_ZIP_PATH, directory_to_extract_to="3rdparty")
+
+    clean_and_build_directory(dir_path="coverage")
+    clean_and_build_directory(dir_path="build")
+
+    os.chdir(LAUNCH_DIR + '/build')
+    run_CMake_cmd()
+    compile_project()
+    run_UnitTestRunner(launch_dir=LAUNCH_DIR)
+    os.chdir(LAUNCH_DIR + '/coverage')
+
+    gcovr_whitelist = generate_gcovr_filter(formatted_user_list=format_user_list(user_list=USER_WHITELIST),
+                                            default_list=DEFAULT_WHITELIST)
```

Comment:
nothing wrong here, my preference is to give them all a line, but its up to you:

```
gcovr_whitelist = generate_gcovr_filter(
    formatted_user_list=format_user_list(user_list=USER_WHITELIST),
    default_list=DEFAULT_WHITELIST)
```